### PR TITLE
FINERACT-2539: Add unit tests for MathUtil

### PR DESCRIPTION
**Pull request by devi-pathak2263**
##Description

This PR adds unit tests for the MathUtil utility class located in fineract-core.

The MathUtil class contains several helper methods used across the platform for null handling, arithmetic operations, numeric comparisons, and BigDecimal utilities. These utilities are frequently used by different modules and therefore benefit from dedicated test coverage.

The added tests verify the behavior of multiple helper methods including:

- Null handling utilities (nullToDefault, nullToZero, zeroToNull)
- Numeric comparison helpers (isGreaterThanZero, isLessThanZero)
- Arithmetic helpers (add, subtract, abs)
- BigDecimal helper methods
- Percentage calculation (percentageOf)
- Formatting utilities such as stripTrailingZeros

The tests include validation of typical scenarios as well as important edge cases such as:
- null inputs
- zero values
- negative values
- BigDecimal precision handling

These tests improve reliability and increase the overall test coverage of shared mathematical utilities used throughout the system.

JIRA: https://issues.apache.org/jira/browse/FINERACT-2539

## Testing

Executed locally:

./gradlew spotlessApply
./gradlew :fineract-core:test

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per [our guidelines](https://github.com/apache/fineract/blob/develop/CONTRIBUTING.md#pull-requests)
- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.
- [x] Create/update [unit or integration tests](https://fineract.apache.org/docs/current/#_testing) for verifying the changes made.
- [x] Follow our [coding conventions](https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions).
- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes
- [x] [This PR must not be a "code dump"](https://cwiki.apache.org/confluence/display/FINERACT/Pull+Request+Size+Limit). Large changes can be made in a branch, with assistance. Ask for help on the [developer mailing list](https://fineract.apache.org/#contribute).

Your assigned reviewer(s) will follow our [guidelines for code reviews](https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide).

### CODE REVIEW

File: fineract-core/src/test/java/org/apache/fineract/infrastructure/core/service/MathUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
+/**
```
Comment (vidakovic) [inline fineract-core/src/test/java/org/apache/fineract/infrastructure/core/service/MathUtilTest.java:1]: Why is MathUtil needed? There are already functions on BigInteger/BigDecimal for add/subtract/mulitply/divide...

Comment (devi-pathak2263) [reply fineract-core/src/test/java/org/apache/fineract/infrastructure/core/service/MathUtilTest.java:1]: This PR only adds unit tests for the existing `MathUtil` utility class. The implementation of `MathUtil` was already present in the codebase and is currently used in several parts of the project.

The goal of this PR is to improve test coverage and ensure the correctness of the helper methods already provided by `MathUtil`, especially around null handling, comparison helpers, and convenience wrappers used across the platform.

No changes were made to the implementation itself.

**Comment by adamsaghy** [comment]
@devi-pathak2263 Can you please:

Set the commit email address to match with your github account email address?
Set GPG signature for commit?

**Comment by adamsaghy** [comment]
```
/home/runner/work/fineract/fineract/fineract-core/src/test/java/org/apache/fineract/infrastructure/core/service/IpAddressUtilsTest.java:21: error: [WildcardImport] Wildcard imports, static or otherwise, should not be used
import static org.junit.jupiter.api.Assertions.*;

**Comment by devi-pathak2263** [comment]
> @devi-pathak2263 Can you please:
> 
> Set the commit email address to match with your github account email address? Set GPG signature for commit?

I have updated the commit author email to match my GitHub account and added a signed commit using SSH.

**Comment by adamsaghy** [comment]
@devi-pathak2263 You need to amend the original commit to take effect, but before you do that, please use GPG instead of SSH. SSH would require to maintain `allowedSigners` but we dont want that.

**Comment by devi-pathak2263** [comment]
- Updated the test imports to avoid wildcard usage and comply with project conventions.

- Amended the commit to use a GPG-signed commit (instead of SSH) and aligned the author email with my GitHub account.


Please let me know if anything else needs adjustment.

**Comment by devi-pathak2263** [comment]
Hi maintainers,

The PR is ready and all requested changes have been addressed.
It looks like the workflows are awaiting approval.

Could you please review and approve the workflows so the checks can run?

Thank you!
